### DISCUSSION TITLE
Use generate both node and react

### DISCUSSION THREAD

**Discussion by IRediTOTO** [ivanhofer/typesafe-i18n#375] [Q&A] [answered]
Hi, I am using typesafe-i18n with NextJS.
I want to use both on client and server api, is there any way generate both of Node and React in same time when run yarn typesafe-i18?

**Comment by ivanhofer** [comment, marked as answer]
Hi @IRediTOTO,
currently the generator only supports a single `adapter` option.
But since the generated output doesn't change frequently, you could:

1. run `typesafe-i18n` with the `adapter` option set to `'node'`
2. set the `adapter` to `'react'` and leave it like this

Then you would have the `i18n-node.ts` and `i18n-react.tsx` file in your repo and you can use both at the same time.
The `i18n-node.ts` file itself gets not updated, but it will read the type information from a shared file that gets updated also when using any other adapter.

**Comment by ivanhofer** [comment]
@IRediTOTO you can checkout `typesafe-i18n` version `5.7.0`, which now supports multiple [`adapters`](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/generator#adapters)
